Objective-C/Tests/DatabaseTest.m

```diff
@@ -2421,22 +2421,159 @@ - (void) testDeleteIndex {
+    
+    
+    // index2
+    CBLFullTextIndexConfiguration* config2 = [[CBLFullTextIndexConfiguration alloc] initWithExpression: @[@"detail"]
+                                                                                         ignoreAccents: NO
+                                                                                              language: nil];
+    Assert([c createIndexWithName: @"index2" config: config2 error: &error]);
+    
+    // index3
+    CBLFullTextIndexConfiguration* config3 = [[CBLFullTextIndexConfiguration alloc] initWithExpression: @[@"es_detail"]
+                                                                                         ignoreAccents: YES
+                                                                                              language: @"es"];
+    Assert([c createIndexWithName: @"index3" config: config3 error: &error]);
+    
+    // same index twice!
+    CBLFullTextIndexConfiguration* config4 = [[CBLFullTextIndexConfiguration alloc] initWithExpression: @[@"detail"]
+                                                                                         ignoreAccents: NO
+                                                                                              language: nil];
+    Assert([c createIndexWithName: @"index2" config: config4 error: &error]);
+    
+    // index4: use backtick in case of property with hyphen
+    CBLFullTextIndexConfiguration* config5 = [[CBLFullTextIndexConfiguration alloc] initWithExpression: @[@"`es-detail`"]
+                                                                                         ignoreAccents: YES
+                                                                                              language: @"es"];
+    Assert([c createIndexWithName: @"index4" config: config5 error: &error]);
+    
+    // verify indexes returning them
+    NSArray* names = [c indexes: &error];
+    AssertEqual(names.count, 4u);
+    AssertEqualObjects(names, (@[@"index1", @"index2", @"index3", @"index4"]));
+    
+    // DELETE INDEX
+    Assert([c deleteIndexWithName: @"index1" error: &error]);
+    Assert([c deleteIndexWithName: @"index2" error: &error]);
+    names = [c indexes: &error];
+    AssertEqual(names.count, 2u);
+    AssertEqualObjects(names, (@[@"index3", @"index4"]));
+}
+
+#pragma mark - Temporary API test
+// TODO: remove when implementation and unit tests are done
+// This is generic test to make sure, Collection APIs are working fine for QE.
 - (void) testCollection {
     [self createDocs: 10];
     CBLDocument* doc1 = [self.db documentWithID: $sprintf(@"doc_%03d", 1)];
     
     NSError* error = nil;
-    CBLDatabase* db = [[CBLDatabase alloc] initWithName: @"dbName" error: &error];
-    CBLCollection* c = [db createCollectionWithName: @"collection1" scope: nil error: &error];
+    CBLCollection* c = [self.db createCollectionWithName: @"collection1" scope: nil error: &error];
```

Comment:
What is the purpose of createDocs and get doc1 above?

Reply:
Sorry, this doc1 is used in below `[collection deleteDocument..]` functions we can remove when we are done with that.